### ISSUE TITLE
Social Registration Confined to 1 user type and not the user type of registration form its attached to

### ISSUE THREAD

**Issue by mrbm** [ultimatemember/ultimatemember#137]
Social Registration Confined to 1 user type and not the user type of registration form its attached to

Expected usage would be that a registration form i set up that has a specic user role and that has social registration activated would register users with that specific user role when using the social registration method on that form page...it doesn't.

I was pulling my hair out thinking this was a bug...but it looks like how it is designed. 

Thoughts?

**Comment by champsupertramp** [comment]
Hi @mrbm ,

Have you created a support ticket on https://www.ultimatemember.com/support/ ? 

Regards,

**Comment by mrbm** [comment]
yes, but after going back and forth with alex since the 25th, I realized this is how it is programmed...and it seems that the right way would be to have the member level of the registration form that the social registration is attached to should supersede the social registration default level.

In the file core/um-actions-register.php, line 123, inserted the following hack for my purposes:

```
// ben start
$splitup = explode('&', $_SERVER['REQUEST_URI']);
if('/myregistrationpageformemberlevelexpert/?provider=linkedin' == $splitup[0]) {
    $args['role'] = $args['submitted']['role'] = $role = 'expert';
}
// ben end
```
This temp fix works for me, but obviously not a solution. Hope this input is valuable. Do you agree this is a bug that should be fixed?

-Ben

**Comment by champsupertramp** [comment]
Hi @mrbm, Did you disable the Global Settings of the registration form?

what's your email address? Please send me your license key and I'll send you the pre-release version of UM Social Login which has fixes for this issue.

Regards,

**Comment by mrbm** [comment]
Just replied on my open ticket to have it forwarded to you.

Thanks.

**Comment by mrbm** [comment]
After testing, Pre release Ultimate Member 1.3.60 is still not recognizing the default role level for sign up forms with social registration via linked in.

**Comment by mrbm** [comment]
FOLLOW UP: I created a user role called "Member 2" (before was testing with the default "Member" role built into Ultimate member. Now that I set "Member 2" to the default role for the default registration form, it is respecting the default role on the scenario I had described earlier. Hopefully that helps in finding this bug.

HOWEVER, in this scenario is not redirecting to the profile page of the user he/she was trying to message before creating an account. Even better would be if it redirected to the users page he/she was trying to message AND the message modal window was opened.
